releases-implementation/README.md

```diff
@@ -0,0 +1,415 @@
+- When the app version bump is tested, it is not necessarily tested on top of the latest release, but actually on top of the latest release plus multiple merged, but unreleased changes in the cluster-$provider app.
+- After the app version bump is merged, it may be released right away, but it may also take some time until it is released, depending on the current state of the cluster-$provider app repo and what unreleased changes it has.
+- Here the team that owns the app is dependant on the provider-integration team and/or on the provider-independent team, because they are needed for the PR reviews and the new release. This practically means that the team that owns the app does not fully own the delivery of the app. More cross-team reviews, more cross-team communication required, more friction.
+
+Now let’s see how the above scenario would look like with the releases repository.
+- The team that owns the app opens a new PR in the releases repo where they create one or more new releases.
+- In the simplest case, app version bump is the only change in the new releases, so it is tested and delivered fully independently.
+- In case there is an existing/draft PR in the release repo where new releases are being added with new versions of other apps, that existing PR can be updated.
+	- If there is a need to have the new app version tested and delivered ASAP, e.g. in case of CVEs, the existing/draft releases repo PR does not have to be updated, and another PR can be created specifically for the new app version (after this the previously created PR can be rebased/updated).
+- The delivery of the new app versions is not entangled with the delivery of cluster-$provider app Helm chart changes.
+- The team that owns the app can create the new release on their own and they do not need to ask other teams for the review/approval, since the unit tests in the releases repository verify that the newly created releases respects the defined release rules, e.g. new app patch delivered to all providers, new app minor must be delivered in a new minor release, etc.
+
+Changes to the current e2e tests would be minimal initially.
+- Testing of cluster-$provider app would require one addition, which would be:
+	- Looking up the latest release and setting the release label on the App CR.
+	- Like today, cluster-test-suites would use custom version of cluster-$provider app from the branch it is testing (in which case the mutating webhook would not set the cluster-$provider app version based on the Release CR).
+- The existing app testing where app versions are overridden via ephemeral Helm values would continue working as is.
+- As a future improvement, cluster-test-suites could run all tests not only for the latest release, but for all releases which use the same major cluster-$provider app version.
```

Comment:
Tests would also now need to be triggered from the `releases` repo if teams are able to bump app versions there. Each of those draft PRs for new releases need to be tested. This would require a completely new pipeline in Tekton.

Reply:
Yes, I had that on my mind, just forgot to write it, thanks for the reminder, will add that as well.